src/deploy/s3-upload.js

```diff
@@ -107,7 +107,8 @@ module.exports = {
 
         const callback = (err, data) => {
             if (err) {
-                console.error(`s3 upload of ${bucket}/${key} failed, error:`, err);
+                console.error(`s3 upload of ${bucket}/${key} failed, error: ${err}`);
+                throw err;
```

Comment:
how does the log look now?

Reply:
Here's the simulated error we discussed where I hardcode in an invalid bucket. The stack trace is still appearing because we're throwing the error, but at least it's less verbose given your suggestion to break after the first failure

```
Deploying to: branch-test/skyline/feature_ctv-3647_fix-s3-upload-log
uploading ./dist
uploading file: qa-media.truex.com/branch-test/skyline/feature_ctv-3647_fix-s3-upload-log/index.html ...
uploading file: qa-media.truex.com/branch-test/skyline/feature_ctv-3647_fix-s3-upload-log/main.7b650478dee199b41762.js ...
uploading file: qa-media.truex.com/branch-test/skyline/feature_ctv-3647_fix-s3-upload-log/main.7b650478dee199b41762.js.map ...
uploading file: qa-media.truex.com/branch-test/skyline/feature_ctv-3647_fix-s3-upload-log/main.css ...
uploading file: qa-media.truex.com/branch-test/skyline/feature_ctv-3647_fix-s3-upload-log/main.css.map ...
s3 upload of doesnotexist.truex.com/branch-test/skyline/feature_ctv-3647_fix-s3-upload-log/main.7b650478dee199b41762.js.map failed, error: NoSuchBucket: The specified bucket does not exist
/Users/isaiah/workspace/tar/desktop/truex-shared-js/node_modules/aws-sdk/lib/request.js:31
            throw err;
            ^

NoSuchBucket: The specified bucket does not exist
    at Request.extractError (/Users/isaiah/workspace/tar/desktop/truex-shared-js/node_modules/aws-sdk/lib/services/s3.js:710:35)
    at Request.callListeners (/Users/isaiah/workspace/tar/desktop/truex-shared-js/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/Users/isaiah/workspace/tar/desktop/truex-shared-js/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/Users/isaiah/workspace/tar/desktop/truex-shared-js/node_modules/aws-sdk/lib/request.js:686:14)
    at Request.transition (/Users/isaiah/workspace/tar/desktop/truex-shared-js/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/Users/isaiah/workspace/tar/desktop/truex-shared-js/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /Users/isaiah/workspace/tar/desktop/truex-shared-js/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/Users/isaiah/workspace/tar/desktop/truex-shared-js/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/Users/isaiah/workspace/tar/desktop/truex-shared-js/node_modules/aws-sdk/lib/request.js:688:12)
    at Request.callListeners (/Users/isaiah/workspace/tar/desktop/truex-shared-js/node_modules/aws-sdk/lib/sequential_executor.js:116:18) {
  code: 'NoSuchBucket',
  region: null,
  time: 2022-04-12T22:35:27.194Z,
  requestId: '6TQK5T7F1BFVG8Q7',
  extendedRequestId: 'z3DOnt4yQ43nI0dLuDrjiIZlB82lDACRWcA7v5rZfcMViZv5OsTBuKFNRV5HkH83syuQlWPBdi4=',
  cfId: undefined,
  statusCode: 404,
  retryable: false,
  retryDelay: 91.60358348979823
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! Skyline@1.7.8 deploy-branch: `node ./deploy/deploy.js deploy`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the Skyline@1.7.8 deploy-branch script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/isaiah/.npm/_logs/2022-04-12T22_35_27_214Z-debug.log
```

Reply:
this is fine. a single final stack track doesn't overwhelm things
and the error nicely interrupts everything

Reply:
Agreed! Thanks for the guidance here